### PR TITLE
Add gas estimation

### PR DESCRIPTION
**Pull request by Kubuxu**

### CODE REVIEW

File: scripts/dev/sminer-init [hunk 1]
```diff
@@ -7,4 +7,4 @@ export TRUST_PARAMS=1
 tag=${TAG:-debug}
 
 go run -tags=$tag ./cmd/lotus wallet import ~/.genesis-sectors/pre-seal-t01000.key
-go run -tags=$tag ./cmd/lotus-miner init --actor=t01000 --genesis-miner --pre-sealed-sectors=~/.genesis-sectors --pre-sealed-metadata=~/.genesis-sectors/pre-seal-t01000.json
+go run -tags=$tag ./cmd/lotus-storage-miner init --actor=t01000 --genesis-miner --pre-sealed-sectors=~/.genesis-sectors --pre-sealed-metadata=~/.genesis-sectors/pre-seal-t01000.json
```
Comment (mostcute) [inline scripts/dev/sminer-init:10]: whai is this? is it "lotu-storage-miner" right?

**Comment by mostcute** [comment]
take a look https://github.com/filecoin-project/lotus/pull/2484/files#r457593367

File: scripts/dev/sminer-init (hunk 1 above)
Comment (Kubuxu) [reply scripts/dev/sminer-init:10]: The binary is `lotus-miner` bur directory is `lotus-storage-miner`:
 https://github.com/filecoin-project/lotus/tree/next/cmd

File: api/api_full.go [hunk 2]
```diff
@@ -143,8 +153,8 @@ type FullNode interface {
 	MpoolGetNonce(context.Context, address.Address) (uint64, error)
 	MpoolSub(context.Context) (<-chan MpoolUpdate, error)
 
-	// MpoolEstimateGasPrice estimates what gas price should be used for a
-	// message to have high likelihood of inclusion in `nblocksincl` epochs.
+	// MpoolEstimateGasPrice is depracated
+	// Deprecated: use GasEstimateGasPrice instead
 	MpoolEstimateGasPrice(ctx context.Context, nblocksincl uint64, sender address.Address, gaslimit int64, tsk types.TipSetKey) (types.BigInt, error)
```
Comment (magik6k) [inline api/api_full.go:158]: Can probably just remove (in a followup PR)

Comment (Kubuxu) [reply api/api_full.go:158]: Yeah, I didn't want to bump API version just before we reset.

**Pull request merged**